common/src/main/java/io/netty/util/concurrent/AbstractScheduledEventExecutor.java

```diff
@@ -205,7 +216,7 @@ public void run() {
 
     final void removeScheduled(final ScheduledFutureTask<?> task) {
         if (inEventLoop()) {
-            scheduledTaskQueue().remove(task);
+            scheduledTaskQueue().removeTyped(task);
```

Comment:
one alternative (which the JDK ScheduledExecutorService does) is that it marks the task as failed but leaves the task in place.  This results in an O(1) time op, but allows the queue to possibly get large.  

Since your change is already trading memory for cpu, it may be worth doing.  The downside is that an uncleared queue will eat memory and possibly get promoted.

Reply:
I think the fixed, small overhead of the queue index is better than having weak bounds on the queue size by not removing futures. For example, with a server serving 1K qps, a user setting a request timeout of 10s would have a queue with 10K items at all times, while reconfiguring to 20s would have a queue with 20K items at all times - this doesn't seem like good, predictable behavior.

Also, leaving it in would not really be `O(1)` but just delay the `O(log n)` to the future when the task gets unscheduled, as removal from the front of a queue is always `log n` bubbles. Perhaps this could be worked around with some batch 'remove from front' algorithm, but doesn't seem worth the complexity IMO.

Reply:
I would prefer to keep the current behavior for this PR, and consider alternative behavior in a followup PR.